packages/relay/src/lib/eth.ts

```diff
@@ -1630,6 +1613,61 @@ export class EthImpl implements Eth {
           .then();
       }
     }
+
+    // After the try-catch process above, the `formattedTransactionId` is potentially valid in only two scenarios:
+    //   - The transaction was successfully submitted and fully processed by CN and MN.
+    //   - The transaction encountered "SDK timeout exceeded" or "Connection Dropped" errors from the SDK but still potentially reached the consensus level.
+    // In both scenarios, polling the MN is required to verify the transaction's validity before return the transaction hash to clients.
+    if (formattedTransactionId) {
+      try {
+        const contractResult = await this.mirrorNodeClient.repeatedRequest(
+          this.mirrorNodeClient.getContractResult.name,
+          [formattedTransactionId, requestDetails],
+          this.mirrorNodeClient.getMirrorNodeRequestRetryCount(),
+          requestDetails,
+        );
+
+        if (!contractResult) {
+          if (
+            sendRawTransactionError instanceof SDKClientError &&
+            (sendRawTransactionError.isConnectionDropped() || sendRawTransactionError.isTimeoutExceeded())
+          ) {
+            throw sendRawTransactionError;
+          }
+
+          this.logger.warn(
+            `${requestIdPrefix} No matching transaction record retrieved: transactionId=${submittedTransactionId}`,
+          );
+
+          throw predefined.INTERNAL_ERROR(
+            `No matching transaction record retrieved: transactionId=${submittedTransactionId}`,
+          );
+        }
```

Comment:
Imo, this logic should be inside the if-statement in the `catch` block on line 1597. We don't want to be slowing down all `eth_sendRawTransaction` requests with this polling, only when we get the "connection dropped" and "timeout exceeded" errors:

```typescript
} catch (e: any) {
  // new logic should be only here
  if (e instanceof SDKClientError && (e.isConnectionDropped() || e.isTimeoutExceeded())) {
    const formattedTransactionId = formatTransactionId(e.transactionId);
    const contractResult = await this.mirrorNodeClient.repeatedRequest(
       this.mirrorNodeClient.getContractResult.name,
       [formattedTransactionId, requestDetails],
       this.mirrorNodeClient.getMirrorNodeRequestRetryCount(),
       requestDetails,
     );
     if (!contractResult?.hash) {
       throw e;
     }
     return contractResult.hash;
  }
  // rest of the `catch` logic remains the same as in the `main` branch
}
// rest of the sendRawTransaction logic remains the same as in the `main` branch
```

Reply:
I agree with Victor and also currently the method is even more confusing to me. Maybe as stated above if we move the new logic to the catch it will be slightly better to read

Reply:
Yeah great point! NGL but that was actually my initial thought as well to only include it here, and as you mentioned, leaving it here in the catch block would only allow for mirror node polling when we encounter "connection dropped" or "timeout exceeded" errors. However, the previous implementation also polled the mirror node for transaction validity in successful cases (see [this block](https://github.com/hashgraph/hedera-json-rpc-relay/blob/main/packages/relay/src/lib/eth.ts#L1592-L1609)) to make sure the transaction is fully processed by CN and MN before the transaction hash is returned to the client. 

This PR expands the polling logic to support both the old and new logic without losing any functionality. That’s why I’ve moved it outside of the try-catch-finally block to enable the polling logic to be reused for both successful cases and Timeout or Connection Dropped scenarios.

Reply:
We can query the transaction receipt directly from the consensus node free of cost (and way faster), I see no reason to poll MN in positive scenarios 🤷‍♂️

Reply:
> We can query the transaction receipt directly from the consensus node free of cost (and way faster), I see no reason to poll MN in positive scenarios 🤷‍♂️

Great idea indeed! This is actually a valuable addition to #3202 to enhance `eth_sendRawTransaction`. However, for the scope of this PR, I aim to maintain the original design while expanding it to accommodate the new behavior.

Reply:
@victor-yanev noted this idea down in [this comment](https://github.com/hashgraph/hedera-json-rpc-relay/issues/3202#issuecomment-2465234136) in the #3202 ongoing discussion. If you have any other idea on how to improve e2e latency of the sendRawTransaction, please few free to edit and add more to [the comment](https://github.com/hashgraph/hedera-json-rpc-relay/issues/3202#issuecomment-2465234136)